src/env/__libc_start_main.c

```diff
@@ -78,93 +79,75 @@ init_aux(size_t *auxv_base, char *pn)
 
 void __init_libc(char **envp, char *pn)
 {
-    size_t i, *auxv, aux[AUX_CNT] = {0};
+    size_t i, *auxv, aux[AUX_CNT] = { 0 };
     __environ = envp;
-    for (i = 0; envp[i]; i++)
-        ;
+    for (i=0; envp[i]; i++);
     libc.auxv = auxv = init_aux((void *)(envp + i + 1), pn);
-    for (i = 0; auxv[i]; i += 2)
-        if (auxv[i] < AUX_CNT)
-            aux[auxv[i]] = auxv[i + 1];
-    __hwcap = aux[AT_HWCAP];
-    __sysinfo = aux[AT_SYSINFO];
-    libc.page_size = aux[AT_PAGESZ];
-
-    if (!pn)
-        pn = (void *)aux[AT_EXECFN];
-    if (!pn)
-        pn = "";
-    __progname = __progname_full = pn;
-    for (i = 0; pn[i]; i++)
-        if (pn[i] == '/')
-            __progname = pn + i + 1;
-
-    __init_ssp((void *)aux[AT_RANDOM]);
-
-    if (aux[AT_UID] == aux[AT_EUID] && aux[AT_GID] == aux[AT_EGID] && !aux[AT_SECURE])
-        return;
-
-    struct pollfd pfd[3] = {{.fd = 0}, {.fd = 1}, {.fd = 2}};
-    int r =
+	for (i=0; auxv[i]; i+=2) if (auxv[i]<AUX_CNT) aux[auxv[i]] = auxv[i+1];
```

Comment:
Is this still static-only code?  If not, we probably are at the point where we need to set up an ELF aux arg vector.

Reply:
Yes and No. This function is called twice.
When this is invoked from __dls3 -> crtjmp, this is reading the auxv entries from the stack.

Due to getenv calls in lkl_start_init, this function is also called before loading the kernel. At that point, the auxv entries are fetched from a `elf64_stack` buffer which is setup in `__prepare_elf_stack`.

Reply:
Can you open an issue to track removing the getenv calls from before libc startup?  When we parse the environment variables from the config file, we should pull the SGX-LKL ones into globals and skip them, then we wouldn't need that extra complexity here.